src/output/csv.c

```diff
@@ -476,7 +478,7 @@ static void dump_saved_values(struct context *ctx, GString **out)
 			if (ctx->time && !ctx->sample_rate) {
 				g_string_append_printf(*out, "0%s", ctx->value);
 			} else if (ctx->time) {
-				sample_time_dbl = ctx->out_sample_count++;
+				sample_time_dbl = ctx->out_sample_count - 1;
```

Comment:
You're correct that the sample count needs to be incremented earlier, but why does the `- 1` happen at all?